BadHabbit/BadHabbit/UserStories/SingUpScreen/SignUpViewController.swift

```diff
@@ -0,0 +1,173 @@
+//
+//  SignUpViewController.swift
+//  BadHabbit
+//
+//  Created by Viktoria Misiulia on 19/07/2023.
+//
+
+import UIKit
+import SnapKit
+
+protocol SignUpViewProtocol: AnyObject {
+    var startButton: UIButton { get }
+}
+
+final class SignUpViewController: BaseViewController, SignUpViewProtocol {
+    private lazy var backgroundImageView = UIImageView()
+    private lazy var welcomeLabel = UILabel()
+    private lazy var appNameLabel = UILabel()
+    private lazy var logoImageView = UIImageView()
+    private lazy var logoLabel = UILabel()
+    private lazy var phraseLabel = UILabel()
+    
+    lazy var startButton = UIButton()
+    
+    private let presenter: SignUpScreenPresenterProtocol
+    
+    init(presenter: SignUpScreenPresenterProtocol) {
+        self.presenter = presenter
+        super.init()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        setupUI()
+    }
+    
+    private func setupUI() {
+        configureBackground()
+        
+        addWelcomeLabel()
+        addAppNameLabel()
+        addLogoImageView()
+        addPhraseLabel()
+        addStartButton()
+        
+        setupViewConstraints()
+    }
+    
+    private func configureBackground() {
+        view.backgroundColor = UIColor(hexNumber: 0xCFCFCF)
```

Comment:
If you define color like you will not have adaptive dark theme. Better to define in assets

Reply:
darkBackground static constant was aded in UIColor extension

Reply:
Colors assets catalog was added